lib/src/validator/pubspec_typo.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import '../entrypoint.dart';
+import '../levenshtein.dart';
+import '../validator.dart';
+
+/// Validates that a package's pubspec does not contain any typos in its keys.
+class PubspecTypoValidator extends Validator {
+  PubspecTypoValidator(Entrypoint entrypoint) : super(entrypoint);
+
+  @override
+  Future validate() async {
+    final fields = entrypoint.root.pubspec.fields;
+
+    if (fields == null) return;
+
+    for (final key in fields.keys) {
+      var bestLevenshteinRatio = 100.0;
+      var closestKey = '';
+
+      for (final validKey in _validPubspecKeys) {
+        /// Use a ratio to account allow more more typos in strings with
+        /// longer lengths.
+        final ratio =
+            levenshteinDistance(key, validKey) / (validKey.length + key.length);
+        if (ratio < bestLevenshteinRatio) {
+          bestLevenshteinRatio = ratio;
+          closestKey = validKey;
+        }
+      }
+
+      /// 0.21 is a magic value determined by looking at the most common typos
+      /// in all the pubspecs on pub.dev.
+      if (bestLevenshteinRatio > 0 && bestLevenshteinRatio < 0.21) {
+        warnings.add('"$key" is not a key recognized by pub - '
+            'did you mean "$closestKey"?');
+      }
+
+      /// Differs from the above because we allow exact matches here too.
+      if (bestLevenshteinRatio < 0.21 &&
+          (closestKey == 'author' || closestKey == 'authors')) {
+        warnings.add('The "$closestKey" key is deprecated - Use a verified '
+            'publisher (https://dart.dev/tools/pub/verified-publishers) '
+            'instead.');
+      }
```

Comment:
Don't we have another thing that warns about `authors` and `author` being deprecated?

So now users with `author` key will get 2 warnings... and users with `authoR` will get 2 warnings.
Maybe we just drop this part?

Reply:
I don't think users with `authoR` will get two warnings though, because it's checked with an exact match in `validator/deprecated_fields`, so this should only trigger the typo warning. 

I'm open to both removing this part, or adding a `bestLevenshteinRatio > 0` to ensure that it only picks up typos - let me know!